#1033 Migrating GEOADD and GEODIST command

YuvrajGosain opened the pull request:
Issue: #1033 

 - [x] Migrated the evalXXX function with the latest definition
 - [x] Update or add unit tests for the new implementation.
 - [x] All unit tests pass successfully.
 - [x] Ensure all integration tests pass successfully.
 - [x] Ensure integration tests are added for the migrated command on multi-threaded resp server.
 - [x] Move Integration tests for the respective commands under the RESP integration tests directory from Async directory
 - [x] Please validate that the documentation for the respective commands is up to date. If not then consider adding them.

lucifercr07 reviewed (changes requested):
@YuvrajGosain can follow sample doc [template](https://github.com/DiceDB/dice/blob/c3400e88c8f647392840f1caa83c69c635c1bee7/docs/sample_command_docs.md) and update for these commands accordingly it'd be helpful.

lucifercr07 commented:
@YuvrajGosain please resolve conflicts once.

YuvrajGosain commented:
Hi @AshwinKul28 , @lucifercr07 , Apologies for the late reply , was busy in some personal stuff and wasn't able to check the status for this, I'll rebase the PR and add docs for these commands by today.

YuvrajGosain commented:
Hi @AshwinKul28 , @lucifercr07 , I have rebased this branch with the master and added the docs for both the commands as well.

helios2003 commented on docs/src/content/docs/commands/GEOADD.md:
Hi, @YuvrajGosain, there are typos in the spellings of longitude and latitude, found them in Line 54 and Line 58  :)

YuvrajGosain replied:
Hi @helios2003 , this should be resolved now :)

c-harish commented on docs/src/content/docs/commands/GEOADD.md:
```diff
@@ -0,0 +1,92 @@
+
+Here are a few examples demonstrating the usage of the GEOADD command:
+
+### Example 1: Adding new member to a set
```
Drop the numbering of examples as 1, 2, etc. You can refer to the sample doc [here](https://github.com/DiceDB/dice/blob/master/docs/command_docs_template.md).

YuvrajGosain replied:
Hi @c-harish , I have addressed this comment.

c-harish commented on docs/src/content/docs/commands/GEODIST.md:
```diff
@@ -0,0 +1,72 @@
+
+Here are a few examples demonstrating the usage of the GEODIST command:
+
+### Example 1: Adding new member to a set
```
Same here.

YuvrajGosain replied:
Hi @c-harish , I have addressed this comment.

c-harish commented on integration_tests/commands/http/geo_tests.go:
```diff
@@ -0,0 +1,86 @@
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
```
Please shift to testify for assertions. You can refer to other test files for example.

YuvrajGosain replied:
Hi @c-harish , I have addressed this comment.

c-harish commented on integration_tests/commands/resp/geo_tests.go:
```diff
@@ -0,0 +1,86 @@
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
```
Please shift to testify for assertions. You can refer to other test files for example.

YuvrajGosain replied:
Hi @c-harish , I have addressed this comment.

c-harish commented on integration_tests/commands/websocket/geo_tests.go:
```diff
@@ -0,0 +1,87 @@
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Please shift to testify for assertions. You can refer to other test files for example.

JyotinderSingh commented:
@YuvrajGosain Please address the remaining review comments

YuvrajGosain commented:
> @YuvrajGosain Please address the remaining review comments

@JyotinderSingh Hi , I have addressed the comments and rebased the branch with master. Pls have a look

JyotinderSingh commented on docs/src/content/docs/commands/GEOADD.md:
```diff
@@ -0,0 +1,92 @@
+
+When the GEOADD command is issued, DiceDB performs the following steps:
+
+1. It checks if argument count is valid or not. If not an error is thrown.
+2. It checks whether optional parameters are applied or not.
```
These points are not required.

JyotinderSingh commented on docs/src/content/docs/commands/GEODIST.md:
```diff
@@ -0,0 +1,72 @@
+description: The `GEODIST` command in Redis is used to calculate the distance between two members (geospatial points) stored in a geospatial index(set). 
+---
+
+The `GEODIST` command in Redis is used to calculate the distance between two members (geospatial points) stored in a geospatial index(set).
```
```suggestion
The `GEODIST` command in DiceDB is used to calculate the distance between two members (geospatial points) stored in a geospatial index(set).
```

JyotinderSingh commented on docs/src/content/docs/commands/GEODIST.md:
```diff
@@ -0,0 +1,72 @@
+
+When the GEODIST command is issued, DiceDB performs the following steps:
+
+1. It checks if argument count is valid or not. If not an error is thrown.
```
this point is not needed.

JyotinderSingh reviewed (changes requested):
Thanks for this massive migration effort @YuvrajGosain! The changes mostly look good, just had a few minor comments.

JyotinderSingh commented on docs/src/content/docs/commands/GEOADD.md:
```diff
@@ -0,0 +1,92 @@
+
+When the GEOADD command is issued, DiceDB performs the following steps:
+
+1. It checks if argument count is valid or not. If not an error is thrown.
+2. It checks whether optional parameters are applied or not.
+3. It checks whether longitude and latitude are valid or not. If not an error is thrown. 
+3. It checks whether the set exists or not.
+4. If set doesn't exist new set is created or else the same set is used. 
+5. It adds or updates the member in the set.
+6. It returns number of members added. 
```
```suggestion
1. It checks whether longitude and latitude are valid or not. If not an error is thrown. 
2. It checks whether the set exists or not.
3. If set doesn't exist new set is created or else the same set is used. 
4. It adds or updates the member in the set.
5. It returns number of members added. 
```

JyotinderSingh commented on docs/src/content/docs/commands/GEODIST.md:
```diff
@@ -0,0 +1,72 @@
+
+When the GEODIST command is issued, DiceDB performs the following steps:
+
+1. It checks if argument count is valid or not. If not an error is thrown.
+2. It gets the sorted set(key).
+3. It gets the scores(geohashes) from the sorted sets for both the members.
+4. It calculates the distance bw them and returns it.
```
```suggestion
1. It gets the sorted set(key).
2. It gets the scores(geohashes) from the sorted sets for both the members.
3. It calculates the distance bw them and returns it.
```

YuvrajGosain commented:
Hi @JyotinderSingh , just realized that I also pushed the changes onto the branch simultaneously and saw it being merged :)
